### PR TITLE
Feature - ThienLuong - MapView

### PR DESCRIPTION
**Pull request by at-thienluong**
- [x] Completed MapView

### CODE REVIEW

File: BaiTap06/ThienLuongQ/CollectionView/AppDelegate.swift [hunk 1]
```diff
@@ -63,4 +71,59 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+            stopStarndardLocationServices()
+        case .authorizedAlways, .authorizedWhenInUse:
+            enableLocationServices()
+            startStandardLocationServices()
+        case .notDetermined:
+            break
+        }
+    }
+    
+    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
```
Comment (at-bienle) [inline BaiTap06/ThienLuongQ/CollectionView/AppDelegate.swift]: Delete unused code

File: BaiTap06/ThienLuongQ/CollectionView/ExLoginHomeDetail/ExLoginHomeDetail/Master/MapViewController/MapAnnotation.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    var imageName: String = ""
+    
+    init(indentify: String, title: String, subtitle: String, coord: CLLocationCoordinate2D, imageName: String) {
+        self.indentify = indentify
+        self.title = title
+        self.subtitle = subtitle
+        self.coordinate = coord
+        self.imageName = imageName
+        
+        super.init()
```
Comment (at-bienle) [inline BaiTap06/ThienLuongQ/CollectionView/ExLoginHomeDetail/ExLoginHomeDetail/Master/MapViewController/MapAnnotation.swift]: Move `super.init()` to top.

File: BaiTap06/ThienLuongQ/CollectionView/ExLoginHomeDetail/ExLoginHomeDetail/Manager/ModelData.swift [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+//  CollectionView
+//
+//  Created by PCI0013 on 1/9/19.
+//  Copyright © 2019 PCI0013. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+final class ModelData {
```
Comment (at-bienle) [inline BaiTap06/ThienLuongQ/CollectionView/ExLoginHomeDetail/ExLoginHomeDetail/Manager/ModelData.swift]: Rename class

File: BaiTap06/ThienLuongQ/CollectionView/ExLoginHomeDetail/ExLoginHomeDetail/Master/MapViewController/MapScreenViewController.swift [hunk 4]
```diff
@@ -11,8 +11,259 @@ import MapKit
...
+    
+    // MARK: - Life cycle
     override func viewDidLoad() {
         super.viewDidLoad()
         title = "Map"
+        
+        AppDelegate.shared.configLocationService()
+        viewModel.loadData()
+        
+        mapView.showsUserLocation = true
```
Comment (at-bienle) [inline BaiTap06/ThienLuongQ/CollectionView/ExLoginHomeDetail/ExLoginHomeDetail/Master/MapViewController/MapScreenViewController.swift]: Please move all line config mapView in to `configMapView()` function

File: BaiTap06/ThienLuongQ/CollectionView/ExLoginHomeDetail/ExLoginHomeDetail/Master/MapViewController/MapScreenViewController.swift [hunk 5]
```diff
@@ -11,8 +11,259 @@ import MapKit
...
+    func mapView(_ mapView: MKMapView, annotationView view: MKAnnotationView, calloutAccessoryControlTapped control: UIControl) {
+        let stores = viewModel.getStoreModelForItem()
+        let detailViewController = DetailViewController()
+        detailViewController.viewModel = DetailViewModel(store: stores[view.tag])
+        navigationController?.pushViewController(detailViewController, animated: true)
+    }
+}
+
+// MARK: - UIGestureRecognizerDelegate
+extension MapScreenViewController: UIGestureRecognizerDelegate {
```
Comment (at-bienle) [inline BaiTap06/ThienLuongQ/CollectionView/ExLoginHomeDetail/ExLoginHomeDetail/Master/MapViewController/MapScreenViewController.swift]: Delete unused code

File: BaiTap06/ThienLuongQ/CollectionView/ExLoginHomeDetail/ExLoginHomeDetail/Master/MapViewController/MapScreenViewController.swift [hunk 6]
```diff
@@ -11,8 +11,259 @@ import MapKit
...
+        mapView.removeOverlays(mapView.overlays)
+        let pickupCoord = mapView.userLocation.coordinate
+        guard let destinationCoord = view.annotation?.coordinate else { return }
+        showRouteOnMap(pickupCoordinate: pickupCoord, destinationCoordinate: destinationCoord)
+    }
+    
+    func mapView(_ mapView: MKMapView, annotationView view: MKAnnotationView, calloutAccessoryControlTapped control: UIControl) {
+        let stores = viewModel.getStoreModelForItem()
+        let detailViewController = DetailViewController()
+        detailViewController.viewModel = DetailViewModel(store: stores[view.tag])
```
Comment (at-bienle) [inline BaiTap06/ThienLuongQ/CollectionView/ExLoginHomeDetail/ExLoginHomeDetail/Master/MapViewController/MapScreenViewController.swift]: Define in viewModel function get DetailViewModel(store: stores[view.tag])

File: BaiTap06/ThienLuongQ/CollectionView/ExLoginHomeDetail/ExLoginHomeDetail/Master/MapViewController/MapScreenViewController.swift (hunk 5 above)
Comment (at-thienluong) [reply BaiTap06/ThienLuongQ/CollectionView/ExLoginHomeDetail/ExLoginHomeDetail/Master/MapViewController/MapScreenViewController.swift]: extension có dùng ở trên a

**Pull request merged**